controllers/api/pwa.js

```diff
@@ -193,4 +196,33 @@ router.get('/:id*?', (req, res) => {
   });
 });
 
+router.post('/add', bodyParser.json(), (req, res) => {
+  const idToken = req.body.idToken;
+
+  if (!idToken) {
+    res.sendStatus(401, JSON.stringify({error: 'user not logged in'}));
+    return;
+  }
+
+  const manifestUrl = req.body.manifestUrl;
+  if (!manifestUrl) {
```

Comment:
I think it would be also good to check that this is a proper URL.

Reply:
updated.